docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -10,26 +10,40 @@ This document contains canister development best practices for both Motoko and R
 
-#### Recommendation
+Furthermore, the controller of canisters serving web content (such as e.g. the [asset canister](../../references/asset-canister)) could maliciously modify the web application to e.g. steal user funds or perform security sensitive actions on the user's behalf. For example, if [Internet Identity](../../developer-docs/integrations/internet-identity/overview) is used, the user principal's private key for the given origin is stored in the browser storage, and a malicious app can therefore fully control the private key and the user's session and any assets controlled by that key.  
+
+#### Recommendations
+
+In the following list, we first provide recommendations for centralized dapp control, and then move to recommendations for increasingly decentralized settings. From a security perspective, more decentralization is favorable. The following list could also be used as a basis for assessing a dapp's level of decentralization. This is just a set of recommendations and may not be complete. 
+1. **The dapp uses central, off-chain components:** The application makes use of centralized components that are e.g. running in the cloud. The owners of these cloud services have full control over the application and assets managed by it. Your application should likely be further decentralized by avoiding central components. But while you have them, [securely manage your keys in the cloud](https://cloudsecurityalliance.org/research/topics/cloud-key-management/). 
+2. **The dapp is controlled by the developer team:** Your project is not under decentralized control, for example because it is in an early development stage or does not (yet) hold significant funds. In that case, it is recommended to manage access to your canisters securely and ideally not letting individuals take full control of the application. To achieve that, consider requiring several individuals or parties to perform any security sentive operations. Such operations are e.g. canister controller operations, or security sensitive changes at the application level that are restricted to privileged principals, such as modifying access controls. A helpful tool to achieve this is the [threshold canister](https://github.com/dfinity/threshold). Manage key material using hardware security modules, such as e.g. [YubyHSM](https://www.yubico.com/ch/store/yubihsm-2-series/) and physically protect these. Some of these devices support threshold signature schemes, which can help to further secure the setup. To increase transparency about the changes made to a dapp, consider using a tool like [LaunchTrail](https://github.com/spinner-cash/launchtrail). 
```

Comment:
@domwoe : 

> @robin-kunzler It'd be great if the team could at some point have a look at https://github.com/spinner-cash/launchtrail since I think it is also an interesting approach between a dapp under centralized control and a DAO.

Thanks for the input, took a brief look at their readme and added a link here, see last sentence in this paragraph. WDYT?

Reply:
@domwoe I suppose this is OK and will resolve this thread so I can merge. If you have any input happy to take it!